test/mongo_test.exs

```diff
@@ -124,6 +124,23 @@ defmodule Mongo.Test do
       Mongo.find(c.pid, coll, %{}, sort: [foo: -1], batch_size: 2, limit: 2) |> Enum.to_list
   end
 
+  test "find_one", c do
+    coll = unique_name()
+
+    assert [] = Mongo.find(c.pid, coll, %{}) |> Enum.to_list
+
+    assert {:ok, _} = Mongo.insert_one(c.pid, coll, %{foo: 42, bar: 1})
+
+    assert nil == Mongo.find_one(c.pid, coll, %{foo: 43})
+    assert %{"foo" => 42} = Mongo.find_one(c.pid, coll, %{}, batch_size: 1)
+
+    assert {:ok, _} = Mongo.insert_one(c.pid, coll, %{foo: 43})
+
+    assert %{"foo" => 42} = Mongo.find_one(c.pid, coll, %{}, batch_size: 1)
+    # should return the first one so the next test fails
+    assert %{"foo" => 43} != Mongo.find_one(c.pid, coll, %{}, batch_size: 1)
```

Comment:
I don't think there's a reason to set `:batch_size` here. In fact, maybe we should force it to `1` in the `find_one`function.